src/renderer/views/preferences.js

```diff
@@ -43,6 +44,29 @@ function renderDownloadDirSelector (state) {
   })
 }
 
+function renderFileHandlers (state) {
+  var definition = {
+    key: 'file-handlers',
+    label: 'Handle Torrent Files'
+  }
+  var buttonText = state.unsaved.prefs.isFileHandler
+    ? 'Remove default app for torrent files'
+    : 'Make WebTorrent the default app for torrent files'
+  var controls = [(
+    <button key='toggle-handlers'
+      className='btn'
+      onClick={toggleFileHandlers}>
+      {buttonText}
+    </button>
+  )]
+  return renderControlGroup(definition, controls)
+
+  function toggleFileHandlers () {
+    var isFileHandler = state.unsaved.prefs.isFileHandler
+    dispatch('updatePreferences', 'isFileHandler', !isFileHandler)
```

Comment:
Should use `setStateValue` to match the other place it's used. Or, better just get rid of that function and call `dispatch('updatePreferences', ...)` directly everywhere.